src/java/org/apache/cassandra/service/ReplicaFilteringProtection.java

```diff
@@ -100,7 +110,9 @@
     ReplicaFilteringProtection(Keyspace keyspace,
                                ReadCommand command,
                                ConsistencyLevel consistency,
-                               InetAddress[] sources)
+                               InetAddress[] sources,
+                               int cachedRowsWarnThreshold,
+                               int cachedRowsFailThreshold)
```

Comment:
I didn't see any reason to couple `ReplicaFilteringProtection` to `DatabaseDescriptor` just to get these values. We don't want them to change in the middle of a query anyway.

Reply:
Note that looking at the client warnings implicitly checks that RFP has occurred.